Framework/include/RunTopTagger.h

```diff
@@ -167,21 +167,54 @@ class RunTopTagger
         // Get reconstructed tops and derive needed variables                            
         const std::vector<TopObject*>& tops = ttr.getTops();
         countTops(tops);
+	
+	// -------------------------------------
+        // -- Calculate DeltaR between 2 tops
+        // -------------------------------------
+	double dR_top1_top2 = -1;
+        if(ntops_ == 2)
+        {
+            dR_top1_top2 = tops.at(0)->p().DeltaR(tops.at(1)->p()); 
+        }
+
+	// ---------------------------------------------------------
+        // -- get vectors of the top mass & eta & pT in the event
+        // ---------------------------------------------------------
+        auto& topsMass = tr.createDerivedVec<double>("topsMass"+myVarSuffix_);
+        for(const auto* t : tops)
+        {
+            topsMass.push_back(t->p().M());
+        }
+
+	auto& topsEta = tr.createDerivedVec<double>("topsEta"+myVarSuffix_);
+	for(const auto* t : tops)
+        {
+            topsEta.push_back(t->p().Eta());
+        }
+	
+	auto& topsPt = tr.createDerivedVec<double>("topsPt"+myVarSuffix_);
+        for(const auto* t : tops)
+        {
+            topsPt.push_back(t->p().Pt());
+        }
+
```

Comment:
Please don't loop through "tops" anymore than you have to. You can put all three for loops above into one.

Reply:
OK. I updated and will push them back to gitHub.